app/src/main/kotlin/com/wire/android/feature/OAuthUseCase.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.feature
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.util.Base64
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultRegistry
+import androidx.activity.result.contract.ActivityResultContracts
+import net.openid.appauth.AppAuthConfiguration
+import net.openid.appauth.AuthState
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationRequest
+import net.openid.appauth.AuthorizationResponse
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.AuthorizationServiceConfiguration
+import net.openid.appauth.ResponseTypeValues
+import net.openid.appauth.browser.BrowserAllowList
+import net.openid.appauth.browser.VersionedBrowserMatcher
+import java.security.MessageDigest
+import java.security.SecureRandom
+
+class OAuthUseCase(context: Context) {
+    private var authState: AuthState = AuthState()
+    private var authorizationService: AuthorizationService
+    private var authServiceConfig: AuthorizationServiceConfiguration = AuthorizationServiceConfiguration(
+        Uri.parse(URL_AUTHORIZATION), Uri.parse(URL_TOKEN_EXCHANGE), null, Uri.parse(URL_LOGOUT)
+    )
+    private var appAuthConfiguration: AppAuthConfiguration = AppAuthConfiguration.Builder().setBrowserMatcher(
+        BrowserAllowList(
+            VersionedBrowserMatcher.CHROME_CUSTOM_TAB, VersionedBrowserMatcher.SAMSUNG_CUSTOM_TAB
+        )
+    ).build()
+
+    init {
+        authorizationService = AuthorizationService(context, appAuthConfiguration)
+    }
```

Comment:
why `var`s not `val`s????
and you can merge initialization of `authorizationService`